### ISSUE TITLE
Allow for custom types

### ISSUE THREAD

**Issue by yakir12** [JuliaDiffEq/LabelledArrays.jl#37]
It would be great if this worked with custom types as well:
```julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.0.2 (2018-11-08)
 _/ |\__'_|_|_|\__'_|  |  
|__/                   |

julia> using LabelledArrays

julia> struct A
           x
           end

julia> a = @LArray map(A, [1,2,3]) (:a,:b,:c);
ERROR: UndefVarError: A not defined
Stacktrace:
 [1] top-level scope at /home/yakir/.julia/packages/LabelledArrays/Y2gSe/src/larray.jl:117
```

**Comment by ChrisRackauckas** [comment]
`a = @LArray [A(1),A(2),A(3)] (:a,:b,:c)` should work. What you wrote there isn't equivalent because has a runtime calculation instead of the values in there so it doesn't know how to parse it.

**Comment by yakir12** [comment]
Hmm:
```julia
julia> a = @LArray [A(1), A(2), A(3)] (:a,:b,:c);
ERROR: UndefVarError: A not defined
```
or 
```julia
julia> a1 = A(1)
A(1)

julia> a2 = A(2)
A(2)

julia> a = @LArray [a1, a2] (:a,:b);
ERROR: UndefVarError: a1 not defined
```

**Comment by ChrisRackauckas** [comment]
Interesting, it has to be a literal. We can see if we can make that part evaluate differently.

**Comment by ChrisRackauckas** [comment]
Fixed in #49
